### PR TITLE
Make `emscripten_check_blocking_allowed()` no-op in release builds

### PR DESCRIPTION
**Pull request by kleisauke**
This warning should only be raised when linking with `-sASSERTIONS` or `-sALLOW_BLOCKING_ON_MAIN_THREAD=0`, where the latter setting would treat this warning with a higher severity (i.e., causing the program to abort when the main thread is blocked).

Previously, this warning was always raised on web environments when not linking with `-sMINIMAL_RUNTIME`.

Resolves: #18855.

### CODE REVIEW

**Comment by sbc100** [comment]
We might want to dig a little into why this was deliberately done in #9579

**Comment by sbc100** [comment]
> We might want to dig a little into why this was deliberately done in #9579

Actually it looks like it the warn was originally only meant to show when ASSERTIONS were enabled, but this regressed in #10123

File: test/test_browser.py [hunk 1]
```diff
@@ -3809,14 +3809,23 @@ def test_pthread_main_thread_blocking(self, name):
     print('Test that we error if not ALLOW_BLOCKING_ON_MAIN_THREAD')
```
Comment (sbc100) [inline test/test_browser.py:3809]: Can you remove this `print` statement, or turn it into docstring?

**Review by kripken — APPROVED** [review]

File: test/test_browser.py [hunk 2]
```diff
@@ -3809,14 +3809,23 @@ def test_pthread_main_thread_blocking(self, name):
...
     self.btest(test_file('pthread/main_thread_%s.cpp' % name), expected='abort:Blocking on the main thread is not allowed by default.', args=['-O3', '-sUSE_PTHREADS', '-sPTHREAD_POOL_SIZE', '-sALLOW_BLOCKING_ON_MAIN_THREAD=0'])
     if name == 'join':
```
[the remark this answers is not in the document]
Comment (kleisauke) [reply test/test_browser.py]: 33468ae7f819b7129882495f59077924b94cf018

File: test/test_browser.py (hunk 1 above)
Comment (kleisauke) [reply test/test_browser.py]: 33468ae7f819b7129882495f59077924b94cf018

**Comment by kripken** [comment]
This lgtm - @sbc100 did you have further comments? Looks like your previous ones were addressed.

**Review by sbc100 — APPROVED** [review]

**Comment by kripken** [comment]
(Merging in latest main may help with the tests.)

**Pull request merged**